### ISSUE TITLE
Better Default implementation for date data

### ISSUE THREAD

**Issue by sffc** [unicode-org/icu4x#260]
We should build a `Default` implementation for date data based on root.json.

**Comment by sffc** [comment]
Depends on #355

**Comment by dminor** [comment]
@sffc We plan to release 0.3 in about 3 weeks. Do you expect this to be unblocked in time for the 0.3 release, or should we punt to 0.4?

**Comment by sffc** [comment]
I think we should hard-code this into the source file.  This is only last-resort data.  We don't necessarily need to enforce that it stays up-to-date with root.json; that could be another task for later.

**Comment by sffc** [comment]
Let's hold off on this until @zbraniecki's DateTime refactoring lands.

**Comment by sffc** [comment]
We should do this once all the datetime data refactoring is done.

**Comment by robertbastian** [comment]
Do we still want to do this after the decision in https://github.com/unicode-org/icu4x/issues/1494#issuecomment-1159088941?

**Comment by Manishearth** [comment]
I'd say no, and we can close this. @sffc ?

**Comment by sffc** [comment]
Given the additional further discussion in https://github.com/unicode-org/icu4x/issues/2174, we can close this.
